### ISSUE TITLE
There is no step when scrolling brightnes slider

### ISSUE THREAD

**Issue by stsdc** [elementary/wingpanel-indicator-power#30]
When I try to scroll brightnes slider in power indicator, It responds with only two positions of slider: min or max.
I believe that it should change with a specified step.

**Comment by matzipan** [comment]
Can you confirm that when you are perhaps using your keyboard buttons to change this, you can reach intermediate states?

**Comment by stsdc** [comment]
@matzipan Yes, I can reach intermediate states. Also, You can see, that brightness level in notification is different from that one in slider. Slider not moving, when changing brightness with keyboard.

![zrzut ekranu z 2017-05-01 12-52-00](https://cloud.githubusercontent.com/assets/6031763/25578274/419bd1fe-2e6d-11e7-8909-7b2409db39f9.png)

**Comment by matzipan** [comment]
Can you please install `d-feet`. And, under "Session Bus" look for `org.gnome.SettingsDaemon.Power`. There look for the `/org/gnome/SettingsDaemon/Power/Screen` entry.  And under "Methods" you will find two items: "StepUp" and "StepDown". Double click on them and press "Execute" and report back if you see the same problem.

If you do, chances are the bug resides in Gnome Settings Daemon.

**Comment by stsdc** [comment]
When Executing StepDown brightness decreases for 7% and for StepUp increases for 7%. But problem still persists.

**Comment by matzipan** [comment]
Do you have the skills and will to try and debug this yourself? It is pretty hard to tell what the problem might be without access to the hardware. We can point you in the right direction. 

The file which handles this is: https://github.com/elementary/wingpanel-indicator-power/blob/master/src/Widgets/ScreenBrightness.vala

The build instructions are here: https://github.com/elementary/wingpanel-indicator-power/blob/master/README.md

You can use gdb or just print stuff using the `message` Vala function. If you want to print, you will probably have to also run `wingpanel` from command line.

**Comment by capstan1** [comment]
I have the exact same behaviour on my Laptop. 

Fact for my machine that maybe plays a role: Fn-Button-combination for brightness never worked in eOS (the only Fn-combination not working out-of-the-box). 
I had to manually map "xbacklight -dec 10%" / "xbacklight -inc 10%" to a custom shortcut.

**Comment by stsdc** [comment]
@matzipan I'll try to debug in my spare time.

**Comment by stsdc** [comment]
So far I can say, that slider updates it's value only on cursor dragging and when it shows up from Wingpanel. 
It hasn't synchronization when changing brightness from Fn keys.

Here is some dirty code for changing brightness value and slider value on scroll event.
It's my first time with Vala and I have no idea what I'm doing, so scrolling works only on icon
```
public class Power.Widgets.ScreenBrightness : Gtk.Grid {
    private const string DBUS_PATH = "/org/gnome/SettingsDaemon/Power";
    private const string DBUS_NAME = "org.gnome.SettingsDaemon";

    private Gtk.Image image;
    private Gtk.Scale brightness_slider;
    private Services.DBusInterfaces.PowerSettings iscreen;

    construct {
        orientation = Gtk.Orientation.HORIZONTAL;
        column_spacing = 6;
        init_bus.begin ();

        var image_box = new Gtk.EventBox ();
        image = new Gtk.Image.from_icon_name ("brightness-display-symbolic", Gtk.IconSize.DIALOG);
        image.margin_start = 6;
        image_box.halign = Gtk.Align.START;
        image_box.add (image);
        attach (image_box, 0, 0, 1, 1);

        brightness_slider = new Gtk.Scale.with_range (Gtk.Orientation.HORIZONTAL, 0, 100, 10);
        brightness_slider.margin_end = 12;
        brightness_slider.hexpand = true;
        brightness_slider.draw_value = false;
        brightness_slider.width_request = 175;

        brightness_slider.value_changed.connect (() => {
            on_scale_value_changed.begin ();
        });

      #if OLD_GSD
        brightness_slider.set_value (iscreen.get_percentage ());
        print ("OLD_GSD %i\n", iscreen.get_percentage ());
      #else
        brightness_slider.set_value (iscreen.brightness);
        info ("Screen brightness %i\n", iscreen.brightness);
      #endif

        attach (brightness_slider, 1, 0, 1, 1);

        // delegate all scroll events to the scale
        image_box.scroll_event.connect (on_scroll);
        brightness_slider.scroll_event.connect (on_scroll);

    }

    public void update_slider (int step = 0) {
      #if OLD_GSD
        brightness_slider.set_value (iscreen.get_percentage ());
      #else
        brightness_slider.set_value (iscreen.brightness + step);
        info ("Set screen brightness %i\n", iscreen.brightness);
      #endif
    }

    private bool on_scroll (Gdk.EventScroll event) {
        // brightness_slider.scroll_event (event);
        info ("Scroll screen brightness %i\n", event.direction);
        if (event.direction == 0) {
            update_slider (12);
        } else if (event.direction == 1){
            update_slider (-12);
        }

        return Gdk.EVENT_STOP;
}

    private async void init_bus () {
        try {
            iscreen = Bus.get_proxy_sync (BusType.SESSION, DBUS_NAME, DBUS_PATH, DBusProxyFlags.GET_INVALIDATED_PROPERTIES);
        } catch (IOError e) {
            warning ("screen brightness error %s", e.message);
        }
    }

    //  slider mouse dragging
    private async void on_scale_value_changed () {
        int val = (int) brightness_slider.get_value ();
        try {
          #if OLD_GSD
            if (iscreen.get_percentage () != val) {
                iscreen.set_percentage (val);
            }
          #else
            if (iscreen.brightness != val) {
                iscreen.brightness = val;
                info ("Set screen brightness *** %i\n", iscreen.brightness);
            }
          #endif
        } catch (IOError e) {
            warning ("screen brightness error %s", e.message);
        }
    }
}

```

**Comment by markodp** [comment]
I noticed the same issue. I made some tests and it seems like Gtk Scale multiplies step value by 10.

It should by default step by 10% but for me it actually steps by 100%. I changed step property to 1 and then scrolling on slider changed value by 10% as it should. Setting step to 2 changed value by 20%, 5 by 50, and so on...

I created other sliders inside indicator and they behaved the same way. I reproduced same issue in session indicator. I am not sure why sliders in any other app or inside volume indicator seem to work just fine.
